### PR TITLE
Dead silicons drop clothing

### LINKED ISSUE #10634: Deconstructing borgs doesnt drop clothing items.
### Labels

[BUG][MINOR]

### Description

Deconstructing borgs doesnt drop clothing items.

### Steps to reproduce

1. Be a cyborg
2. Wear some clothes such as a wig
3. Remove your own head or make a nother person do it.
4. All limbs drop but clothes/wig do not.

### Expected Behavior

When a borg is deconstructed they dont drop any clothing items they were wearing.

### Additional Information & Screenshots

Reported by: IPingu
On server: Goonstation 1 Classic: Heisenbee
Round log date: 2022-09-04-09-18
Reported on: 2022-09-04 09:48:48

### PR DESCRIPTION
**Pull request by Bartimeus973**
[C-bug]
## About the PR 
Dead silicons will now drop all their clothing instead of having the clothes disappear into the aether.

## Why's this needed? 
Clothing should not magically disappear. Also fixes #10634

## Changelog

```changelog
(u)Bartimeus973
(+)Cyborgs now drop their clothing on death.
```

### CODE REVIEW

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -260,6 +260,11 @@
 			src.remove_syndicate("death")
 		src.borg_death_alert()
 		src.eject_brain(fling = TRUE) //EJECT
+		if (length(src.clothes))
+			for (var/A in src.clothes)
+				var/obj/item/cloth_item = src.clothes[A]
+				if (cloth_item)
+					cloth_item.set_loc(src.loc)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:267]: how would this be null?

Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:267]: why not just look for obj item as anything in clothes?

Comment (Bartimeus973) [reply code/mob/living/silicon/robot.dm:267]: removing the null chef, it's indeed unnecessary

Comment (Bartimeus973) [reply code/mob/living/silicon/robot.dm]: changed it, and it works on the /kill command, but appears to not work when a cyborg is killed by blunt damage, not sure why. I'll check it out tomorrow

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -260,6 +260,11 @@
 			src.remove_syndicate("death")
 		src.borg_death_alert()
 		src.eject_brain(fling = TRUE) //EJECT
+		if (length(src.clothes))
+			for (var/obj/item/A as anything in src.clothes)
+				src.clothes[A].set_loc(src.loc)
+		//Clear list as it is of static type
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:266]: No, it's not?

File: code/mob/living/silicon/robot.dm [hunk 3]
```diff
@@ -260,6 +260,11 @@
 			src.remove_syndicate("death")
 		src.borg_death_alert()
 		src.eject_brain(fling = TRUE) //EJECT
+		if (length(src.clothes))
+			for (var/obj/item/A as anything in src.clothes)
+				src.clothes[A].set_loc(src.loc)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:265]: What no that's not how for loops work

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/silicon/robot.dm [hunk 4]
```diff
@@ -260,6 +260,8 @@
 			src.remove_syndicate("death")
 		src.borg_death_alert()
 		src.eject_brain(fling = TRUE) //EJECT
+		for (var/obj/item/slot as anything in src.clothes)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:263]:
```suggestion
		for (var/slot in src.clothes)
```

File: code/mob/living/silicon/robot.dm [hunk 5]
```diff
@@ -33,7 +33,7 @@
 	var/total_weight = 0
 	var/datum/robot_cosmetic/cosmetic_mods = null
 
-	var/list/clothes = list()
+	var/list/obj/item/clothes = list()
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:36]:
```suggestion
	var/list/clothes = list()
```

**Review by ZeWaka — APPROVED** [review]
didn't know it was associated weirdness like that

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**